src/components/BookForm.js

```diff
@@ -1,45 +1,44 @@
-import { PropTypes } from 'prop-types';
-
-function Books({ title, author }) {
-  return (
-    <tr>
-      <td>
-        title:
-        { title }
-        , author:
-        { author }
-      </td>
-    </tr>
-  );
-}
-
-Books.propTypes = {
-  title: PropTypes.string.isRequired,
-  author: PropTypes.string.isRequired,
-};
+import { useState } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import { add } from '../redux/books/booksSlice';
+import BookList from './BookList';
 
 function BookForm() {
-  const books = [
-    { title: 'Marjori', author: 'Damien', id: 0 },
-    { title: 'Revolte', author: 'Guy de cars', id: 1 },
-    { title: 'black child', author: 'Kamara laye', id: 2 },
-  ];
+  const [titleT, setTitle] = useState('');
+  const [authorT, setAuthor] = useState('');
+  const books = useSelector((state) => state.books.all);
+  const dispatch = useDispatch();
+  const handleSubmit = (event) => {
+    event.preventDefault();
+    if (titleT !== '' && authorT !== '') {
+      dispatch(add({
+        item_id: `item${books.length + 1}`,
+        title: titleT,
+        author: authorT,
+        category: 'Fiction',
+      }));
+    }
+  };
```

Comment:
-[ ] [OPTIONAL] Again, if you notice, most of the forms are usually cleared on submission. Its because of this reason, I would stand on to suggest that you clear the form inputs when the book is added to the list of books.